src/components/InputNewTodo/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import styles from './InputNewTodo.module.css'
+
+type InputNewTodoProps = {
+    todoTitle: string,
+    onChange: (todoTitle: string) => void,
+    onSubmit: (todo: any) => void,
+
+}
+type InputNewTodoState = {
+    value: string
+}
+
+export class InputNewTodo extends React.Component<InputNewTodoProps, InputNewTodoState> {
+    componentDidUpdate(prevProps: Readonly<InputNewTodoProps>, prevState: Readonly<InputNewTodoState>, snapshot?: any) {
+        if (this.props.todoTitle !== prevProps.todoTitle) {
+            this.setState({value: this.props.todoTitle})
+        }
+    }
+
+    handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        this.props.onChange(e.target.value);
+    }
+
+    handleKeyDown = (event: React.KeyboardEvent) => {
```

Comment:
this function does to specific this logic should be on onSubmit. This violates Solid principles